modules/nw-ingress-integrating-route-secret-certificate.adoc

```diff
@@ -0,0 +1,72 @@
+//
+// * ingress/routes.adoc
+
+:_mod-docs-content-type: PROCEDURE
+[id="nw-ingress-integrating-route-secret-certificate_{context}"]
+= Securing route with external certificates in TLS secrets
+
+:FeatureName: Securing route with external certificates in TLS secrets
+include::snippets/technology-preview.adoc[]
+
+[NOTE]
+====
+This feature applies to both edge routes and re-encrypt routes.
+====
+
+.Prerequisites
+
+. Make sure you enable the `RouteExternalCertificate` feature gate.
+. Make sure you have access to the cluster as a user with the cluster-admin role.
+. Make sure you generate the serving certificate using third-party systems.
+. Make sure you verify that the referenced secret contains a valid TLS certificate and key pair in PEM format of type `kubernetes.io/tls` in the same namespace as the route. Ensure it includes both `tls.key` and `tls.crt` keys.
+
+.Procedure
+
+. Create a `role` in the same namespace as the secret to allow the router service account read access.
++
+[source,terminal]
+----
+$ oc create role secret-reader --verb=get,list,watch --resource=secrets --resource-name=<secret-name> --namespace=<current-namespace> <1> <2>
+----
+<1> Replace <secret-name> with the actual name of your secret.
+<2> Replace <current-namespace> with the namespace where both your secret and route reside.
+
+. Create a `rolebinding` in the same namespace as the secret and bind the router service account to the newly created role.
++
+[source,terminal]
+----
+$ oc create rolebinding secret-reader-binding --role=secret-reader --serviceaccount=openshift-ingress:router --namespace=<current-namespace> <1>
+----
+<1> Replace <current-namespace> with the namespace where both your secret and route reside.
+
+. Create a `route` specifying the `.spec.tls.externalCertificate` field to reference the secret containing your certificate:
++
+.YAML Definition of the Secure Route
+[source,yaml]
+----
+apiVersion: route.openshift.io/v1
+kind: Route
+metadata:
+  name: myedge
+  namespace: test
+spec:
+  host: myedge-test.apps.example.com
+  tls:
+    externalCertificate:
+      name: <secret-name> <1>
+    termination: edge
+    [...]
+[...]
+----
+<1> Replace <secret-name> with the actual name of your secret.
+
+If the secret exists and has a certificate/key pair, the router will serve the generated certificate if all prerequisites are met.
+
+[NOTE]
+====
+If `.spec.tls.externalCertificate` is not provided, the router will use default generated certificates. Users cannot provide both `.spec.tls.certificate`/`.spec.tls.key` and `.spec.tls.externalCertificate`.
```

Comment:
I guess previously, they were in separate points, or at least there was a linebreak in between the 2 sentences. I would suggest keeping them separately.

Reply:
Another thing I observed is that throughout the document, we never used `Users`. Should we replace `Users` with `You` may be?

Reply:
> Another thing I observed is that throughout the document, we never used `Users`. Should we replace `Users` with `You` may be?

I think it's okay to use `You`, as per the OCP doc style.